pkg/network/usm/usm_http2_monitor_test.go

```diff
@@ -869,6 +869,48 @@ func (s *usmHTTP2Suite) TestRawTraffic() {
 				}: 1,
 			},
 		},
+		{
+			name: "Data frame header sent separately from frame payload",
+			// Testing the scenario in which the data frame header is sent separately from the frame payload.
+			messageBuilder: func() [][]byte {
+				headersFrame := newFramer().writeHeaders(t, 1, testHeaders()).bytes()
+				dataFrame := newFramer().writeData(t, 1, true, emptyBody).bytes()
+				headersFrameHeader := append(headersFrame, dataFrame[:9]...)
+				secondMessage := dataFrame[9:]
+				return [][]byte{
+					headersFrameHeader,
+					secondMessage,
+				}
+			},
+			expectedEndpoints: map[usmhttp.Key]int{
+				{
+					Path:   usmhttp.Path{Content: usmhttp.Interner.GetString(http2DefaultTestPath)},
+					Method: usmhttp.MethodPost,
+				}: 1,
+			},
+		},
```

Comment:
Maybe let's send another request in the second message, to ensure we're able to fully capture it?

Reply:
Added one more request after the first data frame, and validate we fully capture it